### PR TITLE
Fixed dupicite key error

### PR DESCRIPTION
**Pull request by matllubos**

### CODE REVIEW

File: auth_token/models.py [hunk 1]
```diff
@@ -40,16 +42,22 @@ def _hash_key(self, key, **kwargs):
...
                 obj = super().create(key=hashed_key, **kwargs)
                 obj.secret_key = key
                 return obj
             except IntegrityError:
-                if attempt > settings.MAX_RANDOM_KEY_ITERATIONS:
-                    raise IntegrityError('Could not produce unique key')
+                if attempt >= settings.MAX_RANDOM_KEY_ITERATIONS:
+                    raise KeyGeneratorError('Could not produce unique key')
+            except PersistenceException as ex:
+                if ex.error_dict and 'key' in ex.error_dict:
```
Comment (matllubos) [inline auth_token/models.py:56]: only if bug is returned for key field. The new key is generated.

File: auth_token/models.py [hunk 2]
```diff
@@ -40,16 +42,22 @@ def _hash_key(self, key, **kwargs):
...
     def create(self, key_generator, **kwargs):
-        for attempt in range(settings.MAX_RANDOM_KEY_ITERATIONS + 1):
+        for attempt in range(1, settings.MAX_RANDOM_KEY_ITERATIONS + 1):
             try:
                 key = key_generator()
                 hashed_key = self._hash_key(key, **kwargs)
                 obj = super().create(key=hashed_key, **kwargs)
                 obj.secret_key = key
                 return obj
             except IntegrityError:
```
Comment (matllubos) [inline auth_token/models.py:52]: there can be two exceptions PersistenceException if key already exists before the save or IntegrityError if key was created into database before chamber validations and DB insert

File: setup.py [hunk 3]
```diff
@@ -27,7 +27,7 @@
         'django>=2.2.14',
         'django-ipware>=3.0.2',
         'import_string==0.1.0',
-        'django-chamber>=0.5.26',
+        'django-chamber>=0.6.5',
```
Comment (matllubos) [inline setup.py:30]: PersistentException.error_dict was added in this version

File: auth_token/models.py (hunk 1 above)
Comment (matllubos) [reply auth_token/models.py:56]: Because I didn't want to hide other errors

File: auth_token/models.py (hunk 2 above)
Comment (rubickcz) [reply auth_token/models.py:52]: ouch...

**Review by rubickcz — APPROVED** [review]

File: auth_token/models.py (hunk 2 above)
Comment (matllubos) [reply auth_token/models.py:52]: yes this was my big mistake, with missing test. Sorry for it

**Pull request merged**